scripts/water-model/water-script.py

```diff
@@ -0,0 +1,203 @@
+    pipeline_id = row.OBJECTID #
+    pipelines_Y = row.Y
+    pipelines_X = row.X
+    scheme_centroids['lat_diff'] = abs(scheme_centroids['Y'] - pipelines_Y)
+    scheme_centroids['lon_diff'] = abs(scheme_centroids['X'] - pipelines_X)
+    scheme_centroids['sqrt'] = np.sqrt(np.square(scheme_centroids['lat_diff'])+np.square(scheme_centroids['lat_diff']))
+    closest_scheme = scheme_centroids[scheme_centroids['sqrt']==scheme_centroids['sqrt'].min()]['DXF_TEXT'].values[0]
+    canals_centroids.loc[it,'DXF_TEXT'] = closest_scheme
+canals_centroids['DXF_TEXT'] = np.where(canals_centroids['Parish']=='Clarendon','MID CLARENDON',canals_centroids['DXF_TEXT'])
+
+canals_centroids_merge = pd.merge(canals_centroids,schemes,on='DXF_TEXT',how='left').drop_duplicates()
+canals_centroids_merge_gb = canals_centroids_merge.groupby(['DXF_TEXT','ACRES'])['LENGTH'].sum().to_frame().reset_index()
+canals_centroids_merge_gb = canals_centroids_merge_gb.rename(columns={'LENGTH':'sum of canal length'})
+canals_centroids_merge_2 = canals_centroids_merge.merge(canals_centroids_merge_gb, on=['DXF_TEXT','ACRES'],how='left').drop_duplicates()
+canals_centroids_merge_2['acres_per_edge'] = canals_centroids_merge_2['ACRES']*canals_centroids_merge_2['LENGTH']/canals_centroids_merge_2['sum of canal length']
+
+canals_centroids_merge_2['acres_per_edge'] = np.where(canals_centroids_merge_2['DXF_TEXT']=='MID CLARENDON',canals_centroids_merge_2['AREA_SERVE'],canals_centroids_merge_2['acres_per_edge'])
+edges_centroids = pd.concat([pipelines_centroids_merge_2,canals_centroids_merge_2])
+#edges_centroids = edges_centroids.dropna(subset=['acres_per_edge'])
+#edges_centroids['acres_per_edge'] = edges_centroids['acres_per_edge'].apply(roundup)
+
+irrigation_assets_NIC_edges = gpd.read_file('/soge-home/projects/mistral/jamaica-ccri/processed_data/networks/water/irrigation_assets_NIC.gpkg', layer ='edges')
+irrigation_assets_NIC_are_egdes = irrigation_assets_NIC.merge(edges_centroids[['edge_id','acres_per_edge','DXF_TEXT']], on='edge_id', how='left').drop_duplicates()
+
+# irrigation_assets_NIC_are_egdes.to_file('irrigation_assets_NIC_are_edges.gpkg', driver='GPKG').drop_duplicates()
+
+#########################################################################
+#########################################################################
+# disruption analysis irrigation
+#########################################################################
+#########################################################################
+
+# damaged_nodes_df = pd.DataFrame([])
+# merged_nodes = pd.merge(wells_merge_2,damaged_nodes_df,on='node_id')
+# merged_nodes_gb = merged_nodes.groupby('DXF_TEXT')['acres_per_node'].sum().reset_index().to_frame()
+
+# damaged_edges_df = pd.DataFrame([])
+# merged_edges = pd.merge(edges_centroids,damaged_edges_df,on='edge_id')
+# merged_edges_gb = merged_edges.groupby('DXF_TEXT')['acres_per_edge'].sum().reset_index().to_frame()
+    
```

Comment:
In order to run a `main` function when this is called as a script, add this:

```suggestion
if __name__ == '__main__':
        main()
```

This also gives a good place to later add configuration or command-line arguments.